### DISCUSSION TITLE
Mod "alternate" should allow you to press any key after a break

### DISCUSSION THREAD

**Discussion by Moltanicaa** [ppy/osu#18013] [Ideas] [answered]
I stumbled upon a 17-second break in the map during the game and by the end of it I don’t remember which key I pressed last. It would be nice to be able to press any key after the break.

The click counter on the bottom right can only help in case of a perfect game, and if I miss at least 1 time, then I will not be able to determine the next key from it.

It can also be fixed by a small stroke on the click counter next to the button that is worth clicking.

**Comment by bdach** [comment]
This already should be the case, though.

https://github.com/ppy/osu/blob/34457078eb669fdfc0677494881e72d1b1393472/osu.Game.Rulesets.Osu/Mods/OsuModAlternate.cs#L50-L55

**Reply by frenzibyte** [reply]
If it somehow still relies on the key before the break period then do link the beatmap, as it might indicate something worse.

**Reply by integerrr** [reply]
It can happen if a button is clicked once before using the same button to click a hitobject right after the break time ends, and it's the only way I can reliably "reproduce" it, if you can call that reproducing at all.

https://user-images.githubusercontent.com/7279624/165857706-40305514-c06c-4f08-98a1-1b0da08d88f0.mp4

I guess this is somewhat relevant as some players do spam input keys during break time for lots of different reasons, this might happen if one of the spam keys just so happened to land after the break time ends

**Reply by frenzibyte** [reply]
Actually, is this about the break time ending specifically? and not that pressing any key while in break causes this issue to happen?

I don't think we want the conditional in that `IsBreakTime` callback, the user should be able to press whatever during the break period, and it should be reset once it ends.

**Reply by bdach** [reply]
that should already be the case too

https://github.com/ppy/osu/blob/34457078eb669fdfc0677494881e72d1b1393472/osu.Game.Rulesets.Osu/Mods/OsuModAlternate.cs#L58-L61

note that the linked method is also responsible for storing the last pressed action, so if that last action isn't stored, then any press is permitted

**Reply by integerrr** [reply]
I've tried to spam keys *during* break time but as far as i can see it doesnt cause this issue, i also tried to hold the key used for the last object before entering break time and that didnt cause any issues as well

**Reply by frenzibyte** [reply]
Ah I see the issue now. Ideally, `OsuModAlternate` should start storing actions after the first object in every section. Will look into this.

**Comment by frenzibyte** [comment, marked as answer]
Should be resolved in https://github.com/ppy/osu/pull/18017
